_posts/2021-05-14-Spring-Boot-TestConfiguration.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Spring Boot @TestConfiguration Annotation
+categories: [spring-boot]
+date: 2021-05-15 00:00:00 +0530
+modified: 2021-05-15 06:00:00 +1000
+excerpt: "Create custom beans or override specific bean behaviors during unit testing of Spring Boot applications "
+image:
+ auto: 0035-switchboard
+---
+In this article, we will see how to create custom beans or override specific beans during unit testing by using the `@TestConfiguration` annotation.
+
+{% include github-project.html url="https://github.com/thombergs/code-examples/tree/master/spring-boot/spring-boot-testconfiguration" %}
+
+## Introducing @TestConfiguration
+
+We use the `@TestConfiguration` annotation during unit testing of Spring Boot applications for creating custom beans and/or overriding the behavior of specific beans.
+
+Let us see an example ...
+
```

Comment:
Having a code example to demonstrate what you are asking at this place will be overwhelming and potentially confusing too; this would also create redundant code listing. As we have a code listing that is broken into pieces and put in a logical place.